eng/common/scripts/Verify-ChangeLog.ps1

```diff
@@ -1,38 +1,42 @@
 # Wrapper Script for ChangeLog Verification
 param (
-    [String]$ChangeLogLocation,
-    [String]$VersionString,
-    [string]$PackageName,
-    [string]$ServiceName,
-    [string]$RepoRoot,
-    [ValidateSet("net","java","js","python")]
-    [string]$Language,
-    [string]$RepoName,
-    [boolean]$ForRelease=$False
+  [String]$ChangeLogLocation,
+  [String]$VersionString,
+  [string]$PackageName,
+  [string]$ServiceName,
+  [string]$RepoRoot,
+  [ValidateSet("net", "java", "js", "python")]
+  [string]$Language,
+  [string]$RepoName,
+  [boolean]$ForRelease = $False
 )
 
 $ProgressPreference = "SilentlyContinue"
 . (Join-Path $PSScriptRoot SemVer.ps1)
 Import-Module (Join-Path $PSScriptRoot modules ChangeLog-Operations.psm1)
 
 $validChangeLog = $false
-if ($ChangeLogLocation -and $VersionString)
+if ($ChangeLogLocation -and $VersionString) 
 {
   $validChangeLog = Confirm-ChangeLogEntry -ChangeLogLocation $ChangeLogLocation -VersionString $VersionString -ForRelease $ForRelease
 }
-else 
+else
 {
   Import-Module (Join-Path $PSScriptRoot modules Package-Properties.psm1)
   if ([System.String]::IsNullOrEmpty($Language))
   {
-    $Language = $RepoName.Substring($RepoName.LastIndexOf('-') + 1)
+    if ($RepoName -match "azure-sdk-for-(?<lang>[^-]+)")
+    {
+      $Language = $matches["lang"]
+    }
   }
```

Comment:
If language isn't set then the below steps will fail in some unknown way so I suggest adding a check here with a clear error message if language isn't set.